pkg/commands/init.go

```diff
@@ -70,8 +94,68 @@ func NewInitCommand(c types.Args, ioStreams cmdutil.IOStreams) *cobra.Command {
 			if err != nil {
 				return err
 			}
-			o.IOStreams.Info("App Deployed Succeed")
-			//TODO(wonderflow) Wait for app running, and print trait info such as route, domain
+
+			tInit := time.Now()
+			sInit := spinner.New(spinner.CharSets[14], 100*time.Millisecond,
+				spinner.WithColor("green"),
+				spinner.WithFinalMSG(""),
+				spinner.WithHiddenCursor(true),
+				spinner.WithSuffix(color.New(color.Bold, color.FgGreen).Sprintf(" %s", "Initializing ...")))
+			sInit.Start()
+		TrackInitLoop:
+			for {
+				time.Sleep(2 * time.Second)
+				if time.Since(tInit) > initTimeout {
+					ioStreams.Info(red.Sprintf("\n%sInitialization Timeout After %s!", emojiTimeout, duration.HumanDuration(time.Since(tInit))))
+					ioStreams.Info(red.Sprint("Please make sure oam-core-controller is installed."))
+					sInit.Stop()
+					return nil
+				}
+				initStatus, failMsg, err := trackInitializeStatus(context.Background(), o.client, o.workloadName, o.appName, o.Env)
+				if err != nil {
+					return err
+				}
+				switch initStatus {
+				case compStatusInitializing:
+					continue
+				case compStatusInitialized:
+					ioStreams.Info(green.Sprintf("\n%sInitialization Succeed!", emojiSucceed))
+					sInit.Stop()
+					break TrackInitLoop
+				case compStatusInitFail:
+					ioStreams.Info(red.Sprintf("\n%sInitialization Failed!", emojiFail))
+					ioStreams.Info(red.Sprintf("Reason: %s", failMsg))
+					sInit.Stop()
+					return nil
+				}
+			}
+
+			sDeploy := spinner.New(spinner.CharSets[14], 100*time.Millisecond,
+				spinner.WithColor("green"),
+				spinner.WithHiddenCursor(true),
+				spinner.WithSuffix(color.New(color.Bold, color.FgGreen).Sprintf(" %s", "Deploying ...")))
+			sDeploy.Start()
+		TrackDeployLoop:
+			for {
+				time.Sleep(2 * time.Second)
+				deployStatus, failMsg, err := trackDeployStatus(context.Background(), o.client, o.workloadName, o.appName, o.Env)
+				if err != nil {
+					return err
+				}
+				switch deployStatus {
+				case compStatusDeploying:
+					continue
+				case compStatusDeployed:
+					ioStreams.Info(green.Sprintf("\n%sDeployment Succeed!", emojiSucceed))
+					sDeploy.Stop()
+					break TrackDeployLoop
+				case compStatusDeployFail:
+					ioStreams.Info(red.Sprintf("\n%sDeployment Failed!", emojiFail))
+					ioStreams.Info(red.Sprintf("Reason: %s", failMsg))
+					sDeploy.Stop()
+					return nil
+				}
+			}
```

Comment:
Please write functions for these check.

Reply:
Write a common function like `LoopCheck` for this loop and check logic is better.

Reply:
Fixed.